docs/source/tutorials-basic/driving.rst

```diff
@@ -0,0 +1,197 @@
+
+=======================
+Describing a Square
+=======================
+
+To make the GoPiGo3 describe a square by moving itself, you need to run the following script.
+To do this, :py:meth:`~easygopigo3.EasyGoPiGo3.drive_cm` and :py:meth:`~easygopigo3.EasyGoPiGo3.turn_degrees`
+methods are required. A square with the side length of 30cm is drawn. The square is drawn clockwise.
+
+.. code-block:: python
+
+   from easygopigo3 import EasyGoPiGo3
+
+   gpg = EasyGoPiGo3()
+   length = 30
+
+   for i in range(4):
+     gpg.drive_cm(length) # drive forward for length cm
+     gpg.turn_degrees(90) # rotate 90 degrees to the right
+
+.. image:: https://i.imgur.com/6Q7uHq9.gif
+
+=====================
+Making Circular Moves
+=====================
+
+Driving straight in one direction is one thing, but rotating around a center axis at a specific radius is
+something entirely different. In this example, the GoPiGo3 draws half a circle and then returns
+on the same track by spinning itself on the spot.
+
+The radius of the circle is set at 50 centimeters and the robot will move for half of the circle (aka 180 degrees).
+
+.. code-block:: python
+
+   from easygopigo3 import EasyGoPiGo3
+
+   gpg = EasyGoPiGo3()
+
+   gpg.orbit(180, 50) # draw half a circle
+   gpg.turn_degrees(180) # rotate the GoPiGo3 around
+   gpg.orbit(-180, 50) # return on the initial path
+   gpg.turn_degrees(180) # and put it in the initial position
+
+.. image:: https://i.imgur.com/cfdbsID.gif
+
+====================
+Drawing an *8* Shape
+====================
+
+Let's say we want to draw an 8 shape with the GoPiGo3 and at the end have the GoPiGo3 reach the same position
+it initially left from.
+
+To do this, we have to use :py:meth:`~easygopigo3.EasyGoPiGo3.orbit` and :py:meth:`~easygopigo3.EasyGoPiGo3.drive_cm` methods.
+
+.. code-block:: python
+
+   from easygopigo3 import EasyGoPiGo3
+
+   gpg = EasyGoPiGo3()
+   radius = 30
+
+   gpg.orbit(-270, radius) # to rotate to the left
+   gpg.drive_cm(radius * 2) # move forward
+   gpg.orbit(270, radius) # to rotate to the right
+   gpg.drive_cm(radius * 2) # move forward
+
+.. image:: https://i.imgur.com/SQrRBd8.gif
+
+=================================
+Going Forward at Increasing Speed
+=================================
+
+In this example, we make the GoPiGo3 go forward at an ever increasing speed. We start of with a speed of ``50`` and end up going at ``300``.
+:py:meth:`~easygopigo3.EasyGoPiGo3.forward`, :py:meth:`~easygopigo3.EasyGoPiGo3.set_speed` and :py:meth:`~easygopigo3.EasyGoPiGo3.stop` methods are used.
+
+.. code-block:: python
+
+   from easygopigo3 import EasyGoPiGo3
+   from time import time, sleep
+
+   gpg = EasyGoPiGo3()
+
+   # setting speed to lowest value and
+   # calculating the step increase in speed
+   current_speed = 50
+   end_speed = 400
+   step = (end_speed - current_speed) / 20
+   gpg.set_speed(current_speed)
+
+   # start moving the robot at an ever increasing speed
```

Comment:
maybe a warning here that the behaviour will only be achieved with the version of the library from Nov 2018 onwards?  People who try this with a previous version of the library will not get the same result..